google-cloud-storage/src/main/java/com/google/cloud/storage/BlobId.java

```diff
@@ -114,6 +120,21 @@ public static BlobId of(String bucket, String name, Long generation) {
     return new BlobId(checkNotNull(bucket), checkNotNull(name), generation);
   }
 
+  /**
+   * Creates a {@code BlobId} object.
+   *
+   * @param storageUrl the Storage url to create the blob from
+   */
+  public static BlobId fromStorageUrl(String storageUrl) {
```

Comment:
Maybe `fromGsutilUri` instead of storage? To my knowledge, there are many different types of storage URLs
* `https://storage.googleapis.com/bucket/blob`
* `https://storage.cloud.google.com/bucket/blob`

In the cloud console these format of URI are labeled `gsutil URI`
![image](https://user-images.githubusercontent.com/1754037/122830837-eb63d000-d2b6-11eb-953e-d73a33f7b836.png)